From 010b699c92a9a6ffdca1874cd2e3c6c054d212e0 Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <perex@perex.cz>
Date: Wed, 10 Dec 2025 12:12:47 +0100
Subject: [PATCH 7/7] conf: HDA-Intel: define pcm configuration block only one
 time

There may be multiple HDA-Intel soundcards in the system. Overwrite
the PCM configurations when loaded multiple times.

Signed-off-by: Jaroslav Kysela <perex@perex.cz>
---
 src/conf/cards/HDA-Intel.conf | 32 ++++++++++++++++----------------
 1 file changed, 16 insertions(+), 16 deletions(-)

diff --git a/src/conf/cards/HDA-Intel.conf b/src/conf/cards/HDA-Intel.conf
index 5451606f4d31..76775b9753df 100644
--- a/src/conf/cards/HDA-Intel.conf
+++ b/src/conf/cards/HDA-Intel.conf
@@ -4,7 +4,7 @@
 
 <confdir:pcm/front.conf>
 
-HDA-Intel.pcm.front.0 {
+HDA-Intel.pcm.front.!0 {
 	@args [ CARD ]
 	@args.CARD {
 		type string
@@ -29,7 +29,7 @@ HDA-Intel.pcm.front.0 {
 }	
 
 # default with dmix+softvol & dsnoop
-HDA-Intel.pcm.default {
+HDA-Intel.pcm.!default {
 	@args [ CARD ]
 	@args.CARD {
 		type string
@@ -84,7 +84,7 @@ HDA-Intel.pcm.surround71.0 cards.HDA-Intel.pcm.front.0
 
 <confdir:pcm/iec958.conf>
 
-HDA-Intel.pcm.iec958.0 {
+HDA-Intel.pcm.iec958.!0 {
 	@args [ CARD AES0 AES1 AES2 AES3 ]
 	@args.CARD {
 		type string
@@ -163,7 +163,7 @@ HDA-Intel.pcm.iec958.0 {
 	hint.device 1
 }
 
-HDA-Intel.pcm.hdmi.common {
+HDA-Intel.pcm.hdmi.!common {
 	@args [ CARD DEVICE CTLINDEX AES0 AES1 AES2 AES3 ]
 	@args.CARD {
 		type string
@@ -212,7 +212,7 @@ HDA-Intel.pcm.hdmi.common {
 	hint.device $DEVICE
 }
 
-HDA-Intel.pcm.hdmi.0 {
+HDA-Intel.pcm.hdmi.!0 {
 	@args [ CARD AES0 AES1 AES2 AES3 ]
 	@args.CARD { type string }
 	@args.AES0 { type integer }
@@ -235,7 +235,7 @@ HDA-Intel.pcm.hdmi.0 {
 	}
 }
 
-HDA-Intel.pcm.hdmi.1 {
+HDA-Intel.pcm.hdmi.!1 {
 	@args [ CARD AES0 AES1 AES2 AES3 ]
 	@args.CARD { type string }
 	@args.AES0 { type integer }
@@ -258,7 +258,7 @@ HDA-Intel.pcm.hdmi.1 {
 	}
 }
 
-HDA-Intel.pcm.hdmi.2 {
+HDA-Intel.pcm.hdmi.!2 {
 	@args [ CARD AES0 AES1 AES2 AES3 ]
 	@args.CARD { type string }
 	@args.AES0 { type integer }
@@ -281,7 +281,7 @@ HDA-Intel.pcm.hdmi.2 {
 	}
 }
 
-HDA-Intel.pcm.hdmi.3 {
+HDA-Intel.pcm.hdmi.!3 {
 	@args [ CARD AES0 AES1 AES2 AES3 ]
 	@args.CARD { type string }
 	@args.AES0 { type integer }
@@ -304,7 +304,7 @@ HDA-Intel.pcm.hdmi.3 {
 	}
 }
 
-HDA-Intel.pcm.hdmi.4 {
+HDA-Intel.pcm.hdmi.!4 {
 	@args [ CARD AES0 AES1 AES2 AES3 ]
 	@args.CARD { type string }
 	@args.AES0 { type integer }
@@ -327,7 +327,7 @@ HDA-Intel.pcm.hdmi.4 {
 	}
 }
 
-HDA-Intel.pcm.hdmi.5 {
+HDA-Intel.pcm.hdmi.!5 {
 	@args [ CARD AES0 AES1 AES2 AES3 ]
 	@args.CARD { type string }
 	@args.AES0 { type integer }
@@ -350,7 +350,7 @@ HDA-Intel.pcm.hdmi.5 {
 	}
 }
 
-HDA-Intel.pcm.hdmi.6 {
+HDA-Intel.pcm.hdmi.!6 {
 	@args [ CARD AES0 AES1 AES2 AES3 ]
 	@args.CARD { type string }
 	@args.AES0 { type integer }
@@ -373,7 +373,7 @@ HDA-Intel.pcm.hdmi.6 {
 	}
 }
 
-HDA-Intel.pcm.hdmi.7 {
+HDA-Intel.pcm.hdmi.!7 {
 	@args [ CARD AES0 AES1 AES2 AES3 ]
 	@args.CARD { type string }
 	@args.AES0 { type integer }
@@ -396,7 +396,7 @@ HDA-Intel.pcm.hdmi.7 {
 	}
 }
 
-HDA-Intel.pcm.hdmi.8 {
+HDA-Intel.pcm.hdmi.!8 {
 	@args [ CARD AES0 AES1 AES2 AES3 ]
 	@args.CARD { type string }
 	@args.AES0 { type integer }
@@ -419,7 +419,7 @@ HDA-Intel.pcm.hdmi.8 {
 	}
 }
 
-HDA-Intel.pcm.hdmi.9 {
+HDA-Intel.pcm.hdmi.!9 {
 	@args [ CARD AES0 AES1 AES2 AES3 ]
 	@args.CARD { type string }
 	@args.AES0 { type integer }
@@ -442,7 +442,7 @@ HDA-Intel.pcm.hdmi.9 {
 	}
 }
 
-HDA-Intel.pcm.hdmi.10 {
+HDA-Intel.pcm.hdmi.!10 {
 	@args [ CARD AES0 AES1 AES2 AES3 ]
 	@args.CARD { type string }
 	@args.AES0 { type integer }
@@ -467,7 +467,7 @@ HDA-Intel.pcm.hdmi.10 {
 
 <confdir:pcm/modem.conf>
 
-HDA-Intel.pcm.modem.0 {
+HDA-Intel.pcm.modem.!0 {
 	@args [ CARD ]
 	@args.CARD {
 		type string
-- 
2.52.0

